src/protocol/ipa_prf/mod.rs

```diff
@@ -98,32 +100,47 @@ where
 
     let prf_key = gen_prf_key(&convert_ctx);
 
-    let pseudonymed_user_ids = ctx
-        .parallel_join(input_rows.iter().zip(repeat(prf_key)).enumerate().map(
-            |(idx, (record, prf_key))| {
-                let convert_ctx = convert_ctx.clone();
-                let eval_ctx = eval_ctx.clone();
-                async move {
-                    let record_id = RecordId::from(idx);
-                    let prf_of_match_key =
-                        convert_to_fp25519::<_, BA64>(convert_ctx, record_id, &record.match_key)
-                            .await?;
-                    eval_dy_prf(eval_ctx, record_id, &prf_key, &prf_of_match_key).await
-                }
-            },
-        ))
+    let pseudonymed_inputs = ctx
+        .parallel_join(
+            input_rows
+                .iter()
+                .zip(zip(
+                    repeat(prf_key),
+                    zip(repeat(convert_ctx), repeat(eval_ctx)),
+                ))
+                .enumerate()
+                .map(
+                    |(idx, (record, (prf_key, (convert_ctx, eval_ctx))))| async move {
+                        let record_id = RecordId::from(idx);
+                        let prf_of_match_key = convert_to_fp25519::<_, BA64>(
+                            convert_ctx,
+                            record_id,
+                            &record.match_key,
+                        )
+                        .await?;
+                        let pseudonym = eval_dy_prf(eval_ctx, record_id, &prf_key, &prf_of_match_key).await?;
+                        Ok(PrfShardedIpaInputRow {
```

Comment:
The compiler is unable to compile this code complaining about unable to resolve E type of Error.

Reply:
```suggestion
                        Ok::<_, Error>(PrfShardedIpaInputRow {
```

yea, compiler trips over this code. I don't see a reason why, but a tiny hint should make it compile

Reply:
Sigh.. I tried to put this in the variable it was assignd to and not here. Thanks for this!!